### PR TITLE
Solve a CUDA deprecation warning

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: tests/regression/array/implementation.cu [hunk 1]
```diff
@@ -38,9 +38,16 @@ namespace gpu_array {
         }
         cudaPointerAttributes attributes;
         auto ret = cudaPointerGetAttributes(&attributes, descriptor->data);
-        if (ret != cudaSuccess || attributes.memoryType != cudaMemoryTypeDevice) {
+        if (ret != cudaSuccess ||
+#if defined(CUDART_VERSION) && CUDART_VERSION < 10000
+            attributes.memoryType != cudaMemoryTypeDevice
+#else
+            (attributes.type != cudaMemoryTypeDevice || attributes.type != cudaMemoryTypeManaged)
```
Comment (lukasm91) [inline tests/regression/array/implementation.cu]: can this be false? I guess it should be `&&`.

**Review by lukasm91 — CHANGES_REQUESTED** [review]

File: tests/regression/array/implementation.cu (hunk 1 above)
Comment (havogt) [inline tests/regression/array/implementation.cu]:
```suggestion
            (attributes.type != cudaMemoryTypeDevice && attributes.type != cudaMemoryTypeManaged)
```

**Comment by havogt** [comment]
launch jenkins

**Review by lukasm91 — APPROVED** [review]

**Pull request merged**